lib/client.js

```diff
@@ -78,6 +81,10 @@ module.exports = function (dependencies) {
               this.config.token.regenerate(tokenGeneration);
               resolve(this.write(notification, device, retryCount + 1));
               return;
+            } else if (status === "500" && response.reason === "InternalServerError") {
+              stream.connection.close();
+              let error = new VError("Error 500, stream ended unexpectedly");
+              resolve({ device, error });
```

Comment:
I think it would be better to return the status code `(resolve({ device, status, response });)` instead of throw error, so that the user can handle this error by reading status code.